cache4.cpp

```diff
@@ -1,4 +1,4 @@
-//cache implementation
+// cache implementation
 #include <iostream>
 #include <unordered_map>
 #include <vector>
```

Comment:
tagWidth = addressWidth - setIndexWidth; // Number of bits for tag

Where this variable "tagWidth" is used?

    // extract tag bits from address
    int extractTag(int address) {
        return address >> setIndexWidth; // shift to the right to exclude set index bits
    }
    
    how this function "extractTag" is utilized?

Reply:
I left the tagwidth definition by mistake, i tried another method to isolate the tag bits.

extractTag function basically removes the index bits from the address and gives tag bits
i was just trialing a higher level implementation considering tag bits, index bits and offset bits
i overlooked the irrelevant function before uploading.